util/math_reference.h

```diff
@@ -285,10 +285,23 @@ sycl::marray<T, N> select(sycl::marray<T, N> a, sycl::marray<T, N> b,
 
 /* absolute value */
 template <typename T>
-T abs(T x) {
-  return x < 0 ? -x : x;
+sycl_cts::resultRef<T> abs(T x) {
+  using U = typename std::make_unsigned<T>::type;
```

Comment:
```suggestion
  using U = typename std::make_unsigned_t<T>;
```

Reply:
Sure, and `typename` is not needed in that case either. Updated. I tried to go with the existing style but I guess you want to update gradually to the newer forms?